core/engine_test.go

```diff
@@ -968,3 +968,45 @@ func TestMinIterationDuration(t *testing.T) {
 	// But we expect the custom counter to be added to 4 times
 	assert.Equal(t, 4.0, getMetricSum(collector, "testcounter"))
 }
+
+func TestMinIterationDurationInSetupTeardownStage(t *testing.T) {
+	t.Parallel()
+
+	runner, err := js.New(
+		&loader.SourceData{URL: &url.URL{Path: "/script.js"}, Data: []byte(`
+		import { sleep } from "k6";
+
+		export function setup() {
+			sleep(0.5);
+		}
+
+		export let options = {
+			minIterationDuration: "1s",
+			duration: "2s",
+			setupTimeout: "2s",
+		};
+
+		export default function () {
+		};`)},
+		nil,
+		lib.RuntimeOptions{},
+	)
+	require.NoError(t, err)
+
+	engine, err := NewEngine(local.New(runner), runner.GetOptions())
+	require.NoError(t, err)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	errC := make(chan error)
+	go func() { errC <- engine.Run(ctx) }()
+
+	select {
+	case <-time.After(10 * time.Second):
+		cancel()
+		t.Fatal("Test timed out")
+	case err := <-errC:
+		cancel()
+		require.NoError(t, err)
+		require.False(t, engine.IsTainted())
+	}
```

Comment:
I don't think this test tests ... the code , does this fail if the below code is not there ?

Reply:
Sorry, forgot to ammend, the test script should be:

```
                 import { sleep } from "k6";

		export function setup() {
			sleep(1);
		}

		export let options = {
			minIterationDuration: "2s",
			duration: "2s",
			setupTimeout: "2s",
		};

		export default function () {
		};
```

`sleep(0.5)` seems sleep 0 second.